core/node/api_server/src/web3/namespaces/en.rs

```diff
@@ -95,12 +98,21 @@ impl EnNamespace {
             .map_err(DalError::generalize)?
             .context("Genesis batch doesn't exist")?;
 
-        let protocol_version = genesis_batch
+        let minor = genesis_batch
             .protocol_version
             .context("Genesis is not finished")? as u16;
+        let minor = ProtocolVersionId::try_from(minor).context("Malformed protocol version")?;
+        let patch = storage
+            .protocol_versions_dal()
+            .first_vk_patch_for_version(minor)
```

Comment:
why first and not last?

Reply:
In this method we return patch for genesis batch. We don't store mapping (batch number => patch) anywhere so I don't see better option here, note that it only works for **genesis** batch

Reply:
okay